src/core/openthread.cpp

```diff
@@ -977,33 +979,33 @@ void HandleActiveScanResult(void *aContext, Mac::Frame *aFrame)
         memcpy(&result.mExtendedPanId, beacon->GetExtendedPanId(), sizeof(result.mExtendedPanId));
     }
 
-    handler(&result);
+    handler(&result, aClientContext);
 
 exit:
     return;
 }
 
 ThreadError otDiscover(uint32_t aScanChannels, uint16_t aScanDuration, uint16_t aPanId,
-                       otHandleActiveScanResult aCallback)
+                       otHandleActiveScanResult aCallback, void *aCallbackContext)
 {
     return sThreadNetif->GetMle().Discover(aScanChannels, aScanDuration, aPanId, &HandleMleDiscover,
-                                           reinterpret_cast<void *>(aCallback));
+                                           reinterpret_cast<void *>(aCallback), aCallbackContext);
```

Comment:
Then if we save the callback context globally, then it would make sense to save the callback in the same place. In other words, we don't need to pass anything but the `&HandleMleDiscover` down to Mle, right? Same for passing down `&HandleActiveScanResult` to Mac when it calls `ActiveScan`.